script.js

```diff
@@ -1,4 +1,668 @@
+    playerCardMessage =
+      playerCardMessage +
+      `<br> ${allPlayerCards[currentPlayer][aceCounter].name} of ${allPlayerCards[currentPlayer][aceCounter].emojiSuit}`;
+    aceCounter += 1;
+  }
+  // If player busts, output results
+  if (playerPoints > 21) {
+    var bustOutput = "";
+    // However, if player draws ace, allow them to change value
+    counter = allPlayerCards[currentPlayer].length - 1;
+    while (counter < allPlayerCards[currentPlayer].length) {
+      if (allPlayerCards[currentPlayer][counter].name == "Ace") {
+        gameMode = PLAYER_ACTION;
+        return `${userName[currentPlayer]}, you will bust and lose unless you change your ace card's value! <br><br> Type 'ace' to change its value.<br><br> ${playerCardMessage}`;
+      }
+      counter += 1;
+    }
+    // Also, if player busts and has a previous ace card with value 11, allow them to change value
+    secondCounter = 0;
+    while (secondCounter < allPlayerCards[currentPlayer].length) {
+      if (
+        allPlayerCards[currentPlayer][secondCounter].name == "Ace" &&
+        allPlayerCards[currentPlayer][secondCounter].value == 11
+      ) {
+        gameMode = PLAYER_ACTION;
+        return `${userName[currentPlayer]}, you will bust and lose unless you change your ace card's value! <br><br> Type 'ace' to change its value.<br><br> ${playerCardMessage}`;
+      }
+      secondCounter += 1;
+    }
+    currentPlayer += 1;
+    allPlayerOutcomes.push("lose");
+    bankRoll[currentPlayer - 1] =
+      bankRoll[currentPlayer - 1] - userBet[currentPlayer - 1];
+
+    bustOutput = `${
+      userName[currentPlayer - 1]
+    }, you've bust and lost! 😭 <br><br>  ${playerCardMessage} <br> Your points: ${playerPoints}<br><br> Click submit for the next player's turn!<br><br> Your bank $${
+      bankRoll[currentPlayer - 1]
+    } `;
+    // Change to next player / summarize results if all players have played
+    if (currentPlayer == numPlayers) {
+      playerPoints = 0;
+      gameMode = GAME_SUMMARY;
+    } else {
+      playerPoints = 0;
+      gameMode = SELECT_PLAYER;
+    }
+    return bustOutput;
+  }
+  // If player gets blackjack, output results
+  if (playerPoints == 21) {
+    currentPlayer += 1;
+    bankRoll[currentPlayer - 1] =
+      bankRoll[currentPlayer - 1] + userBet[currentPlayer - 1];
+    allPlayerOutcomes.push("win");
+    // Change to next player / summarize results if all players have played
+    if (currentPlayer == numPlayers) {
+      playerPoints = 0;
+      gameMode = GAME_SUMMARY;
+    } else {
+      playerPoints = 0;
+      gameMode = SELECT_PLAYER;
+    }
+    return `${
+      userName[currentPlayer - 1]
+    }, you've gotten blackjack and won! 🥳 <br><br>${playerCardMessage}<br><br> Your bank $${
+      bankRoll[currentPlayer - 1]
+    } <br><br> Click submit for the next player's turn!`;
+  }
+  return `${userName[currentPlayer]}, ${playerCardMessage}  ${endMessage};`;
+};
```

Comment:
great job implementing the game logic, but this function that stretches nearly 100 lines might be too long. It would be better if this could be broken down into smaller functions